srtcore/core.h

```diff
@@ -394,17 +394,32 @@ class CUDT
     duration minNAKInterval() const { return m_tdMinNakInterval; }
     sockaddr_any peerAddr() const { return m_PeerAddr; }
 
+    /// Returns the number of packets in flight (sent, but not yet acknowledged).
+    /// @returns The number of packets in flight belonging to the interval [0; ...)
     int32_t getFlightSpan() const
     {
-        // This is a number of unacknowledged packets at this moment
-        // Note that normally m_iSndLastAck should be PAST m_iSndCurrSeqNo,
+        // m_iSndCurrSeqNo is the sequence number of the latest original packet sent.
+        // m_iSndLastAck is the sequence number of the first unacknowledged packet.
+        // When there are no packets in flight, m_iSndLastAck = incseq(m_iSndCurrSeqNo).
+
+        // Packets sent:
+        // | 1 | 2 | 3 | 4 | 5 |
+        //   ^               ^
+        //   |               |
+        // m_iSndLastAck     |
+        //               m_iSndCurrSeqNo
+        //
+        // In Flight: [m_iSndLastAck; m_iSndCurrSeqNo]
+        //
+        // Normally m_iSndLastAck should be PAST the m_iSndCurrSeqNo,
         // however in a case when the sending stopped and all packets were
         // ACKed, the m_iSndLastAck is one sequence ahead of m_iSndCurrSeqNo.
         // Therefore we increase m_iSndCurrSeqNo by 1 forward and then
         // get the distance towards the last ACK. This way this value may
-        // be only positive or 0.
+        // be only positive as seqlen() includes endpoints.
+        // Finally, we subtract 1 to exclude the increment added earlier.
 
-        return CSeqNo::seqlen(m_iSndLastAck, CSeqNo::incseq(m_iSndCurrSeqNo));
+        return CSeqNo::seqlen(m_iSndLastAck, CSeqNo::incseq(m_iSndCurrSeqNo)) - 1;
```

Comment:
I have an idea here: make a static method that would get sndlastack and sndcurrseqno as parameters.

Then implement `getFlightSpan()` by calling this function and passing these two fields as parameters.

That would allow to use this function in a unit test.

Reply:
Kinda:
```
static int flightspan(int lastack, int curseq)
{
     return CSeqNo::seqlen(lastack, CSeqNo::incseq(curseq)) - 1;
}

int getFlightSpan() { return flightspan(m_iSndLastAck, m_iSndCurrSeqNo); }
```

Reply:
Created a static version of `getFlightSpan(..)`.
However, there is still a potential vulnerability for future changes in messing the order of arguments.
```c++
getFlightSpan(m_iSndLastAck, m_iSndCurrSeqNo);
```

Reply:
But this makes it at least testable, even if with limited capabilities.